File: clients/javascript/src/apis/idempotency.ts
// this file is @generated

import {
    type IdempotencyAbortIn,
    IdempotencyAbortInSerializer,
} from '../models/idempotencyAbortIn';
import {
    type IdempotencyAbortOut,
    IdempotencyAbortOutSerializer,
} from '../models/idempotencyAbortOut';
import {
    type IdempotencyCompleteIn,
    IdempotencyCompleteInSerializer,
} from '../models/idempotencyCompleteIn';
import {
    type IdempotencyCompleteOut,
    IdempotencyCompleteOutSerializer,
} from '../models/idempotencyCompleteOut';
import {
    type IdempotencyStartIn,
    IdempotencyStartInSerializer,
} from '../models/idempotencyStartIn';
import {
    type IdempotencyStartOut,
    IdempotencyStartOutSerializer,
} from '../models/idempotencyStartOut';
import { IdempotencyNamespace } from './idempotencyNamespace';
import { HttpMethod, CoyoteRequest, type CoyoteRequestContext } from "../request";

export class Idempotency {
    public constructor(private readonly requestCtx: CoyoteRequestContext) {}

    public get namespace() {
        return new IdempotencyNamespace(this.requestCtx);
    }

    /** Start an idempotent request */
    public start(
        idempotencyStartIn: IdempotencyStartIn,
        ): Promise<IdempotencyStartOut> {
        const request = new CoyoteRequest(HttpMethod.POST, "/api/v1/idempotency/start");

        request.setBody(
            IdempotencyStartInSerializer._toJsonObject(
                idempotencyStartIn,
            )
        );
        return request.send(
            this.requestCtx,
            IdempotencyStartOutSerializer._fromJsonObject,
        );
    }/** Complete an idempotent request with a response */
    public complete(
        idempotencyCompleteIn: IdempotencyCompleteIn,
        ): Promise<IdempotencyCompleteOut> {
        const request = new CoyoteRequest(HttpMethod.POST, "/api/v1/idempotency/complete");

        request.setBody(
            IdempotencyCompleteInSerializer._toJsonObject(
                idempotencyCompleteIn,
            )
        );
        return request.send(
            this.requestCtx,
            IdempotencyCompleteOutSerializer._fromJsonObject,
        );
    }/** Abandon an idempotent request (remove lock without saving response) */
    public abort(
        key: string,
        idempotencyAbortIn: IdempotencyAbortIn,
    ): Promise<IdempotencyAbortOut> {
        const request = new CoyoteRequest(HttpMethod.POST, "/api/v1/idempotency/abort");

        request.setBody(
            IdempotencyAbortInSerializer._toJsonObject({
                ...idempotencyAbortIn,
                key,
            })
        );
        
        return request.send(
            this.requestCtx,
            IdempotencyAbortOutSerializer._fromJsonObject,
        );
    }
}

File: clients/java/src/main/java/com/svix/diom/apis/Idempotency.java
// this file is @generated
package com.svix.diom.apis;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.svix.diom.ApiException;
import com.svix.diom.HttpClient;
import com.svix.diom.Utils;
import java.io.IOException;
import java.util.Collections;
import java.util.HashMap;
import java.util.Map;
import java.util.Set;
import lombok.Getter;
import okhttp3.Headers;
import okhttp3.HttpUrl;
import com.svix.diom.models.IdempotencyAbortIn;
import com.svix.diom.models.IdempotencyAbortOut;
import com.svix.diom.models.IdempotencyCompleteIn;
import com.svix.diom.models.IdempotencyCompleteOut;
import com.svix.diom.models.IdempotencyStartIn;
import com.svix.diom.models.IdempotencyStartOut;

public class Idempotency {
    private final HttpClient client;

    public Idempotency(HttpClient client) {
        this.client = client;
    }

    /** Start an idempotent request */
    public IdempotencyStartOut start(
        final IdempotencyStartIn idempotencyStartIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/idempotency/start");
        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            idempotencyStartIn,
            IdempotencyStartOut.class
            );
    }

    /** Complete an idempotent request with a response */
    public IdempotencyCompleteOut complete(
        final IdempotencyCompleteIn idempotencyCompleteIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/idempotency/complete");
        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            idempotencyCompleteIn,
            IdempotencyCompleteOut.class
            );
    }

    /** Abandon an idempotent request (remove lock without saving response) */
    public IdempotencyAbortOut abort(
        final IdempotencyAbortIn idempotencyAbortIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/idempotency/abort");
        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            idempotencyAbortIn,
            IdempotencyAbortOut.class
            );
    }
}